### ISSUE TITLE
[Bug]: Disney Dreamlight Valley, crashes with errors.

### ISSUE THREAD

**Issue by kosg2022** [originalnicodr/CinematicUnityExplorer#87]
### Are you on the latest version of CinematicUnityExplorer?

- [X] Yes, I'm on the latest version of CinematicUnityExplorer.

### Which release are you using?

BepInEx Unity IL2CPP CoreCLR

### Which game did this occur on?

Disney Dreamlight Valley - Unity 2023.2.11f1

### Describe the issue.

BepInEx works fine for this game, but Unity Explorer (any version) causes errors. I can't use MelonLoader because it's outdated and doesn't support Unity 2023+. When I load the game, it runs for a few seconds before crashing with many console errors. I’ll provide the log and hope you can help! I used to use it before the Unity update, and I'm devastated that it no longer works!
[LogOutput.log](https://github.com/user-attachments/files/16752959/LogOutput.log)

### Relevant log output

```shell
Github isn't letting me paste the log file so I've attached it to the issue directly. Something is not working correctly I'm not sure.
```

**Comment by originalnicodr** [comment]
Why not use the bepinex version then?
I guess it has to do with melon loader not having support for that version of Unity yet?

If there is a modloader version or wrong mod version problem they give [this modloader](https://github.com/Raicuparta/rai-pal) a try.

**Comment by kosg2022** [comment]
> Why not use the bepinex version then? I guess it has to do with melon loader not having support for that version of Unity yet?
> 
> If there is a modloader version or wrong mod version problem they give [this modloader](https://github.com/Raicuparta/rai-pal) a try.

Thank you for your response!

Yeah Melonloader is very outdated for this Unity version, but Bepinex works fine. What I mean by that is I can use Bepinex, I can make my own mods, but Cinematic Unity Explorer (Or any other version/fork) does NOT work. The logoutput is literally all I have to go off of, if you have any other suggestions or ways to test what the issue could be with CinematicUnityExplorer and this game, I would greatly appreciate it.

That other modloader I have just tried it, it installs bepinex and the original unity explorer which doesn't work (and no other version works either)

It's not an anti-cheat, the game is 99% single player (1% being optional and does not impact gameplay at all) you can play perfectly fine without an internet connection. I've checked the assemblies there's nothing related to anti cheat, tampering etc, and as I've stated earlier, I can write my own mods and they work perfectly fine.

**Comment by kosg2022** [comment]
> > Why not use the bepinex version then? I guess it has to do with melon loader not having support for that version of Unity yet?
> > If there is a modloader version or wrong mod version problem they give [this modloader](https://github.com/Raicuparta/rai-pal) a try.
> 
> Thank you for your response!
> 
> Yeah Melonloader is very outdated for this Unity version, but Bepinex works fine. What I mean by that is I can use Bepinex, I can make my own mods, but Cinematic Unity Explorer (Or any other version/fork) does NOT work. The logoutput is literally all I have to go off of, if you have any other suggestions or ways to test what the issue could be with CinematicUnityExplorer and this game, I would greatly appreciate it.
> 
> That other modloader I have just tried it, it installs bepinex and the original unity explorer which doesn't work (and no other version works either)
> 
> It's not an anti-cheat, the game is 99% single player (1% being optional and does not impact gameplay at all) you can play perfectly fine without an internet connection. I've checked the assemblies there's nothing related to anti cheat, tampering etc, and as I've stated earlier, I can write my own mods and they work perfectly fine.

Just to test things out, I started by replacing the relevant dll's with ones from the latest bepinex bleeding edge / the interop folder, and I noticed that there's a whole bunch of code that doesn't work in the visual studio project once that happens. I inspected the original dll and the "newer" dll and found that yes indeed there are quite some differences... I guess Unity 2023 REALLY changed how things worked and most likely that's the reason this mod doesn't work anymore.

Do you have any guidance as to how to properly update the dll's (did I have the right idea or is there something else / a step that I'm not aware of to make them suitable for the mod?) The code itself is going to need a lot of tweaks, and I'm by no means a professional or close to on this sort of thing. I've made some small mods before and fixed some old broken ones.

**Comment by originalnicodr** [comment]
> Do you have any guidance as to how to properly update the dll's (did I have the right idea or is there something else / a step that I'm not aware of to make them suitable for the mod?) The code itself is going to need a lot of tweaks, and I'm by no means a professional or close to on this sort of thing. I've made some small mods before and fixed some old broken ones.

I am not sure if it's something we have to fix on my end here. As far as I understand, given that the mod depends on the UniverseLib library to draw UI and other Unity-related dlls that expose the engine API so we can interact with it, there might be something missing/not working on the latter ones? Yet you did say that making your own mods worked fine.

I am sorry I can not give more advice about it since my knowledge of the abstraction layer that is the modding API is super small 😞. Pherpas you can ask around in the bepinex or melonloader modding discords about this. They have been able to help me in the past.

**Comment by kosg2022** [comment]
> > Do you have any guidance as to how to properly update the dll's (did I have the right idea or is there something else / a step that I'm not aware of to make them suitable for the mod?) The code itself is going to need a lot of tweaks, and I'm by no means a professional or close to on this sort of thing. I've made some small mods before and fixed some old broken ones.
> 
> I am not sure if it's something we have to fix on my end here. As far as I understand, given that the mod depends on the UniverseLib library to draw UI and other Unity-related dlls that expose the engine API so we can interact with it, there might be something missing/not working on the latter ones? Yet you did say that making your own mods worked fine.
> 
> I am sorry I can not give more advice about it since my knowledge of the abstraction layer that is the modding API is super small 😞. Pherpas you can ask around in the bepinex or melonloader modding discords about this. They have been able to help me in the past.

Yeah it's definitely an issue related to the unity version completely changing how everything works, not everything is accessible the same way (or there's something that needs to be done to the referenced dll's to allow for it to be accessible, I don't know there were a lot of folders in the source code and everything is very vague, where did the dll's come from originally / how to get up to date ones.

I thought you might know considering you're currently maintaining the mod and updating / improving it, but perhaps you haven't experienced the quite drastic change between versions (2023 at the very least) I don't know of many games that have updated to unity 2023. 

The Melonloader discord won't help with anything related to Disney Dreamlight Valley.. not "supported" others saying that it has anti-cheat / anti modding in it, which is complete nonsense when people are still making mods, including myself. The issue with melonloader is just that it's updating at the speed of a snail and they don't support 2023 yet I guess.

Again though Bepinex works just fine for modding.. I guess that the unity explorer / universelib source code and dll's are just too outdated and too much has changed. Would almost need a full rewrite, which is less than ideal and even more so when most games are not affected by this change.
